src/web3py/extensions/tx_utils.py

```diff
@@ -118,9 +120,12 @@ def _sign_and_send_transaction(
         tx_hash = self.w3.eth.send_raw_transaction(signed_tx.rawTransaction)
         logger.info({"msg": "Transaction sent.", "value": tx_hash.hex()})
 
-        tx_receipt = self.w3.eth.wait_for_transaction_receipt(tx_hash)
+        return self._handle_sent_transaction(tx_hash)
 
-        if not tx_receipt:
+    def _handle_sent_transaction(self, transaction_hash: HexBytes) -> Optional[TxReceipt]:
+        try:
+            tx_receipt = self.w3.eth.wait_for_transaction_receipt(transaction_hash)
```

Comment:
`tx_receipt` can still be`None` (by the library code). Let's handle this as well

Reply:
As I see it can't
```
    def wait_for_transaction_receipt(
        self, transaction_hash: _Hash32, timeout: float = 120, poll_latency: float = 0.1
    ) -> TxReceipt:
        try:
            with Timeout(timeout) as _timeout:
                while True:
                    try:
                        tx_receipt = self._transaction_receipt(transaction_hash)
                    except TransactionNotFound:
                        tx_receipt = None
                    if tx_receipt is not None:
                        break
                    _timeout.sleep(poll_latency)
            return tx_receipt

        except Timeout:
            raise TimeExhausted(
                f"Transaction {HexBytes(transaction_hash) !r} is not in the chain "
                f"after {timeout} seconds"
            )
``